odk/src/test/java/org/motechproject/odk/it/EventHandlerIT.java

```diff
@@ -0,0 +1,109 @@
+package org.motechproject.odk.it;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.type.TypeReference;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.motechproject.event.MotechEvent;
+import org.motechproject.odk.constant.EventParameters;
+import org.motechproject.odk.constant.FieldTypeConstants;
+import org.motechproject.odk.constant.OnaConstants;
+import org.motechproject.odk.domain.FormDefinition;
+import org.motechproject.odk.domain.FormElement;
+import org.motechproject.odk.domain.FormInstance;
+import org.motechproject.odk.event.EventHandler;
+import org.motechproject.odk.event.builder.impl.EventBuilderOna;
+import org.motechproject.testing.osgi.container.MotechNativeTestContainerFactory;
+import org.ops4j.pax.exam.ExamFactory;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerSuite;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerSuite.class)
+@ExamFactory(MotechNativeTestContainerFactory.class)
+public class EventHandlerIT extends OdkBaseIT {
+
+    @Inject
+    private EventHandler eventHandler;
+    private List<MotechEvent> events;
+    private FormDefinition formDefinition;
+
+    @Before
+    public void setup() throws Exception {
+
+        formDefinition = getFormDefinitionDataService().byConfigurationNameAndTitle(CONFIG_NAME,TITLE);
+        assertEquals(getFormDefinitionDataService().retrieveAll().size(),1);
+        assertNotNull(formDefinition);
+
+        events = new EventBuilderOna().createEvents(getJson(),formDefinition,getConfiguration());
+        convertEventValues(events);
+    }
+
+
+    @Test
+    public void testPersistFormNestedRepeats() throws Exception{
+        MotechEvent persistEvent = events.get(events.size() - 1);
+        eventHandler.handlePersistForm(persistEvent);
```

Comment:
don't see much point testing like this, since this is fake integration. In the real world we will use ActiveMQ and the event system.